File: jimmy/courses/cpp/base/intToHex.cc
#include <iostream>
using namespace std;

const string HEX = "0123456789ABCDEF";

string intToHex(int x) {
  string hexStr("");
  while (x > 0) {
    hexStr = HEX[x % 16] + hexStr;
    x /= 16;
  }
  return hexStr;
}

int main() {
  string hex = intToHex(2748);
  cout << hex << endl;
  hex = intToHex(1024);
  cout << hex << endl;
  return 0;
}

File: jimmy/courses/java/base/IntToHex.java
class BaseConverter {
  public static final String HEX = "0123456789ABCDEF";

  public static String intToHex(int x) {
    String hexStr = "";
    while (x > 0) {
      hexStr = HEX.charAt(x % 16) + hexStr;
      x /= 16;
    }
    return hexStr;
  }
}

public class IntToHex {
  public static void main(String[] args) {
    String hex = BaseConverter.intToHex(2748);
    System.out.println(hex);
    hex = BaseConverter.intToHex(1024);
    System.out.println(hex);
  }
}